### PR TITLE
MET-87: Add save measurement family internal endpoint + E2E tests

### PR DESCRIPTION
**Pull request by SamirBoulil**
**Description (for Contributor and Core Developer)**

- Add SaveMeasurementFamilyAction Internal endpoints
- Add End 2 End tests

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | ok
| Changelog updated                 | -
| Review and 2 GTM                  | Ok
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/SaveMeasurementFamilyAction.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                'The identifier provided in the route and the one given in the body of the request are different',
+                Response::HTTP_BAD_REQUEST
+            );
+        }
+
+        $decodedRequest = $this->decodeRequest($request);
+        $saveMeasurementFamilyCommand = $this->createSaveMeasurementFamilyCommand($decodedRequest);
+        try {
+            $this->validateCreateMeasurementFamilyCommand($saveMeasurementFamilyCommand);
+            $this->saveMeasurementFamilyHandler->handle($saveMeasurementFamilyCommand);
```
Comment (juliensnz) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/SaveMeasurementFamilyAction.php]: seems weird to call it save. Shouldn't it be called "updateMeasurementFamilyCommand"? Can look like "chipoting" but actually I'm currious about the reason for this naming :)

File: src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/SaveMeasurementFamilyAction.php [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        } catch (\InvalidArgumentException $ex) {
```
Comment (juliensnz) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/SaveMeasurementFamilyAction.php]: so you didn't want to call it $e nor $exception just to create a new convention? ^^

File: src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/SaveMeasurementFamilyAction.php [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            return new JsonResponse(
+                ['code' => Response::HTTP_UNPROCESSABLE_ENTITY, 'message' => $ex->getMessage()],
+                Response::HTTP_UNPROCESSABLE_ENTITY
+            );
+        } catch (ViolationHttpException $ex) {
```
Comment (juliensnz) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/SaveMeasurementFamilyAction.php]: Seems weird to use and catch this exception as it's made to be sent to the user (as every httpExceptions)

File: src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/SaveMeasurementFamilyAction.php (hunk 1 above)
Comment (SamirBoulil) [reply src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/SaveMeasurementFamilyAction.php]: I think we did this with Valentin. It's a create + update so we decided to call it "Save..." which I think is fine

File: src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/SaveMeasurementFamilyAction.php (hunk 3 above)
Comment (SamirBoulil) [reply src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/SaveMeasurementFamilyAction.php]: I thought this too. This is also done this way in Create...Action. We do this to call a ViolationNormalizer which uses the Exception to normalize the violations.

I questionned @quentin about this (why not use the same serializer than AM ?) I think we are following what connectivity is doing and we already have the front type to handle the violations normalized with this exception.
We can change if this is an issue no pb

File: src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/SaveMeasurementFamilyAction.php (hunk 2 above)
Comment (SamirBoulil) [reply src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/SaveMeasurementFamilyAction.php]: ^^ copy / paste i'll fix it

File: src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/SaveMeasurementFamilyAction.php (hunk 3 above)
Comment (tseho) [reply src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/SaveMeasurementFamilyAction.php]: I also said I'm not 100% sure this is the right path and I checked again and its seems Connectivity use another format.
I agree with both of you, it's weird to throw an exception to catch it directly but the violation normalizer take this exact exception in parameter :(

This is a mess and we should think about how to harmonize it.

**Pull request merged**